net/net.go

```diff
@@ -161,8 +161,21 @@ func BigForIP(ip net.IP) *big.Int {
 
 // AddIPOffset adds the provided integer offset to a base big.Int representing a
 // net.IP
-func AddIPOffset(base *big.Int, offset int) net.IP {
-	return net.IP(big.NewInt(0).Add(base, big.NewInt(int64(offset))).Bytes())
+func AddIPOffset(base *big.Int, offset int, isIPv6 bool) net.IP {
+	length := 4
+	if isIPv6 {
+		length = 16
+	}
+	ipBytes := big.NewInt(0).Add(base, big.NewInt(int64(offset))).Bytes()
+	ipBytesLength := len(ipBytes)
+	if ipBytesLength > length {
```

Comment:
is this case possible?

Reply:
You can try extreme scenario like this:
```go
ip := net.ParseIP("255.255.255.0")
res := AddNetIPOffset(ip, 256)
```
Without this line, the underlying byte slice of `res` will be `[1,0,0,0,0]`. Not a valid IP.